File: member/src/main/java/com/openat/member/infrastructure/config/SecurityConfig.java
package com.openat.member.infrastructure.config;

import java.util.List;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.CorsConfigurationSource;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;

/**
 * 인증/인가는 apigateway가 JWT로 이미 끝낸 뒤 X-User-Id/X-User-Roles 헤더로 신뢰해서 넘겨준다
 * ({@code common.auth.UserContextFilter} 참고). 이 서비스가 직접 인증을 다시 수행하지 않으므로,
 * spring-boot-starter-security가 기본으로 켜는 로그인 폼/Basic Auth를 모두 끄고 전부 permitAll 처리한다.
 * (이 설정이 없으면 Spring Security 기본 자동설정 때문에 모든 요청이 401로 막힌다.)
 */
@Configuration
@EnableWebSecurity
public class SecurityConfig {

    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http, CorsConfigurationSource corsConfigurationSource) throws Exception {
        http
                .cors(cors -> cors.configurationSource(corsConfigurationSource))
                .csrf(AbstractHttpConfigurer::disable)
                .httpBasic(AbstractHttpConfigurer::disable)
                .formLogin(AbstractHttpConfigurer::disable)
                .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
                .authorizeHttpRequests(auth -> auth.anyRequest().permitAll());
        return http.build();
    }

    /**
     * apigateway가 호스팅하는 swagger-ui(localhost:8000)에서 member(localhost:9100)를 직접
     * "Try it out"으로 호출할 때 브라우저가 막는 CORS 문제를 풀기 위함. 로컬 개발 전용이라
     * localhost의 모든 포트를 허용 패턴으로 열어둔다 (운영 배포 시에는 실제 프론트 도메인으로 좁혀야 함).
     */
    @Bean
    public CorsConfigurationSource corsConfigurationSource() {
        CorsConfiguration configuration = new CorsConfiguration();
        configuration.setAllowedOriginPatterns(List.of("http://localhost:*", "https://localhost:*"));
        configuration.setAllowedMethods(List.of("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"));
        configuration.setAllowedHeaders(List.of("*"));
        configuration.setAllowCredentials(true);

        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration("/**", configuration);
        return source;
    }

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }
}

File: queue/src/main/kotlin/com/openat/queue/infrastructure/config/SecurityConfig.kt
package com.openat.queue.infrastructure.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity
import org.springframework.security.config.web.server.ServerHttpSecurity
import org.springframework.security.web.server.SecurityWebFilterChain

/**
 * WebFlux+SSE 전환분: MVC의 `SecurityFilterChain`/`HttpSecurity`는 서블릿 전제라 WebFlux
 * 애플리케이션에서는 적용되지 않는다(`spring.main.web-application-type: reactive`로 전환됨).
 * 의도는 예전과 동일하다 - 인증은 apigateway가 단일 지점에서 검증하고, 이 서비스는 게이트웨이가
 * 주입한 헤더(X-User-Id 등)를 신뢰하며 자체 로그인/인가는 하지 않는다. `SecurityWebFilterChain`/
 * `ServerHttpSecurity`가 그 리액티브 대응이다. 세션 미사용(stateless)은 WebFlux 기본값이라
 * MVC 쪽의 `sessionCreationPolicy` 설정에 대응하는 별도 지정이 필요 없다.
 */
@Configuration
@EnableWebFluxSecurity
class SecurityConfig {

    @Bean
    fun securityWebFilterChain(http: ServerHttpSecurity): SecurityWebFilterChain =
        http
            .csrf { it.disable() }
            .httpBasic { it.disable() }
            .formLogin { it.disable() }
            .authorizeExchange { it.anyExchange().permitAll() }
            .build()
}
